### PR TITLE
Replace `pcre` with Rust-implemented regex crate (based on RE2).

### PR DESCRIPTION
**Pull request by BurntSushi**
I fixed the bug in Rust's `regex` library and things seem to be working great now. Even through `regex` is based on RE2, I didn't have to change your regex at all. :-)

I ran this to test:

```bash
make perf^rust
make MAL_IMPL=rust test^mal
```

The first passes and the second passes all tests except for 1, but it looks like the same failing test from #26.

### CODE REVIEW

**Comment by kanaka** [comment]
@BurntSushi I merged it in and tested it but I was getting reader errors. I reverted the change for now since it wasn't working for me. I might have some more time tomorrow to take a look.

**Comment by kanaka** [comment]
@BurntSushi hmm, okay, a fresh checkout of your repo worked. I think mine was checking out a stale version of regex (0.1.17). Probably I forgot to run cargo update. Let me try again.

**Comment by kanaka** [comment]
@BurntSushi yep, it was just me forgetting to run cargo update and getting a stale build of regex. Silly me. I reverted the revert (first time I've done that in git I think) and pushed. We should be all good. 

Thanks for that!

**Comment by BurntSushi** [comment]
Aha! I completely forgot to remind you to run `cargo update`. It has tripped me up more than once, so don't feel bad! :-) Thanks!
